### PR TITLE
Add model for todo entity.

### PR DESCRIPTION
**Pull request by Krishuthapa**
- [x] Add an entity named todo and design a model for it.

### CODE REVIEW

File: src/main/java/test/app/todo/model/Todo.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package test.app.todo;
+
+import javax.persistence.*;
+
+@Entity(name = "todos")
+@Table(schema = "todo")
```
Comment (shradayshakya) [inline src/main/java/test/app/todo/model/Todo.java]: It's a good practice to specify the schema through a constant, such that when the schema changes, changing it through one place would suffice.

File: src/main/java/test/app/todo/model/Todo.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package test.app.todo;
+
+import javax.persistence.*;
```
Comment (shradayshakya) [inline src/main/java/test/app/todo/model/Todo.java]: We don't need to import all the classes from the `javax.persistence` package, import only necessary Classes.

File: src/main/java/test/app/todo/model/Todo.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+@Entity(name = "todos")
+@Table(schema = "todo")
+public class Todo {
+  @Id
+  private Long id;
+
+  private String todoInfo;
```
Comment (shradayshakya) [inline src/main/java/test/app/todo/model/Todo.java:21]: You can specify the column name when generating the table in the database. The auto generated column name in the database might not be reliable.
```
@Column(name = "todo_info")
private String todoInfo;
```
By doing this, we can be confident when writing a raw SQL.

File: src/main/java/test/app/todo/model/Todo.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+  private Boolean isCompleted;
```
Comment (shradayshakya) [inline src/main/java/test/app/todo/model/Todo.java:24]: Here, too specify the column name.
```
@Column(name = "is_completed")
private String isCompleted;
```

File: src/main/java/test/app/todo/model/Todo.java (hunk 3 above)
Comment (Krishuthapa) [reply src/main/java/test/app/todo/model/Todo.java:21]: amazing...

File: src/main/java/test/app/todo/model/Todo.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
+package test.app.todo;
+
+import javax.persistence.*;
+
+@Entity(name = "todos")
+@Table(schema = "todo")
+public class Todo {
+  @Id
+  private Long id;
```
Comment (shradayshakya) [inline src/main/java/test/app/todo/model/Todo.java:18]: When creating a Todo item, you'll have to maintain a counter and explicitly specify the **id** when saving the todo object in the database. 

You can also ask the database to auto-generate the id when creating the todo item, it does this by maintaining its own sequence which we won't need to worry about. 
```
@Id
@GeneratedValue(strategy=GenerationType.AUTO)
private int id;
```

Comment (Krishuthapa) [reply src/main/java/test/app/todo/model/Todo.java:18]: done

File: src/main/java/test/app/todo/model/Todo.java (hunk 4 above)
Comment (Krishuthapa) [reply src/main/java/test/app/todo/model/Todo.java:24]: done got it

**Review by shradayshakya — APPROVED** [review]

**Pull request merged**